ion/cmp_test.go

```diff
@@ -51,8 +51,28 @@ func (thisTimestamp ionTimestamp) eq(other ionEqual) bool {
 	return false
 }
 
-func cmpAnnotations(thisAnnotations, otherAnnotations []string) bool {
-	return reflect.DeepEqual(thisAnnotations, otherAnnotations)
+func cmpAnnotations(thisItem, otherItem ionItem) bool {
+	// Annotation sets are considered equal if the first annotation for each is $ion_symbol_table.
+	// eg. $ion_symbol_table::foo and $ion_symbol_table::bar::baz are considered equal.
+
+	// If the first annotation is $ion_symbol_table, then we want the annotation comparison logic to also apply to
+	// all the inner structs within the ion item.
+	if !thisItem.insideLocalSymbolTable &&
+		len(thisItem.annotations) > 0 && thisItem.annotations[0] == "$ion_symbol_table" {
+		thisItem.setInsideLocalSymbolTable(true)
```

Comment:
It's surprising to me that a method with the signature `cmp(this, other) bool` has stateful side effects.

Reply:
Reworked this to perform the side effects in readCurrentValue() instead.

Reply:
👍